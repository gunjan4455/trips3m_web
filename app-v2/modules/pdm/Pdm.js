import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import { getSelectedPdms } from 'helpers/pdmHelpers';
import { PdmContainer } from './G';

class Pdm extends React.Component {

  static propTypes = {
    setMessage: PropTypes.func.isRequired,
    pdmsData: PropTypes.object.isRequired,
    selectedPdm: PropTypes.string.isRequired,
  };

  static defaultProps = {
    setMessage: () => {
    }
  };

  constructor(props) {
    super(props);
    this.onPdmSelection = this.onPdmSelection.bind(this);
  }

  onPdmSelection(pdm) {
    this.props.setMessage(pdm.message_text);
  }

  render() {
    const { pdmsData, selectedPdm } = this.props;
    const pdmList = getSelectedPdms(pdmsData, selectedPdm);

    if (!pdmList || !pdmList.length) {
      return null;
    }

    return (
      <PdmContainer className="clearfix relative">
        <ul>
          {
            pdmList.map((item, index) =>
              <li key={index} >
                <a className={`at_${index}`} onClick={() => this.onPdmSelection(item)}>
                  {item.pdm_text}
                </a>
              </li>
            )
          }
        </ul>
      </PdmContainer>
    );
  }
}

export default Pdm;

