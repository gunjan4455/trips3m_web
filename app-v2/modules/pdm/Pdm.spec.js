import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import Pdm from './Pdm';

configure({ adapter: new Adapter() });

const DEFAULT_PROPS = {
  setMessage: jest.fn,
  pdmsData: {
    Itinerary: [
      {
        category: 'Itinerary',
        message_enum: 'PDM_CHANGE_ITINERARY_DAYS_ORDER',
        message_text: 'Hey! I want to change the order of itinerary days.',
        pdm_text: 'I want to change the order of itinerary days'
      },
      {
        category: 'Itinerary',
        message_enum: 'PDM_CHANGE_ITINERARY_CITIES',
        message_text: 'Hey! I want to change the cities in the itinerary.',
        pdm_text: 'I want to change the cities in the itinerary'
      },
      {
        category: 'Itinerary',
        message_enum: 'PDM_MODIFY_ITINERARY_PARTICULAR_DAY',
        message_text: 'Hey! I want modification in itinerary of a particular day.',
        pdm_text: 'I want modification in itinerary of a particular day'
      },
      {
        category: 'Itinerary',
        message_enum: 'PDM_MODIFY_ITINERARY_ATTRACTIONS_COVERED',
        message_text: 'Hey! I want modification in the attractions being covered.',
        pdm_text: 'I want modification in the attractions being covered'
      }
    ],
    Hotel: [
      {
        category: 'Hotel',
        message_enum: 'PDM_CHANGE_HOTEL_NEED_BETTER_BUDGET',
        message_text: 'Hey! I need hotel in a better budget.',
        pdm_text: 'Need hotel in a better budget'
      },
      {
        category: 'Hotel',
        message_enum: 'PDM_CHANGE_HOTEL_NEED_BETTER_REVIEWS',
        message_text: 'Hey! I need a hotel with better reviews.',
        pdm_text: 'Need a hotel with better reviews'
      },
      {
        category: 'Hotel',
        message_enum: 'PDM_CHANGE_HOTEL_NEED_HIGHER_ROOM_CATEGORY',
        message_text: 'Hey! I need a hotel with a higher room category.',
        pdm_text: 'Need a hotel with a higher room category'
      },
      {
        category: 'Hotel',
        message_enum: 'PDM_BETTER_ROOM_SAME_HOTEL',
        message_text: 'Hey! I need a better room in the same hotel.',
        pdm_text: 'Need a better room in the same hotel'
      },
      {
        category: 'Hotel',
        message_enum: 'PDM_CHANGE_HOTEL_NEAR_SIGHTSEEING_AREA',
        message_text: 'Hey! I need a hotel near the market/sightseeing areas.',
        pdm_text: 'Need a hotel near the market/sightseeing areas'
      }
    ],
    Inclusion: [
      {
        category: 'Inclusion',
        message_enum: 'PDM_ADD_INCLUSION_CONTACT_ME',
        message_text: 'Hey! I need to get a few inclusions added. Please contact me to discuss further.',
        pdm_text: 'Need to get an inclusion added'
      },
      {
        category: 'Inclusion',
        message_enum: 'PDM_REMOVE_INCLUSION_CONTACT_ME',
        message_text: 'Hey! I need to get a few inclusions removed. Please contact me to discuss further.',
        pdm_text: 'Need to get an inclusion removed'
      },
      {
        category: 'Inclusion',
        message_enum: 'PDM_MODIFY_INCLUSION_CONTACT_ME',
        message_text: 'Hey! I need modifications in a few inclusions. Please contact me to discuss further.',
        pdm_text: 'Need to modify an inclusion'
      }
    ]
  },
  selectedPdm: 'Inclusion'
};

const shallowWrappperPdm = (props) => {
  return shallow(<Pdm {...props} />);
};

describe('Pdm: Render Correctly ', () => {

  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should match with the previous snapshot', () => {
    const wrapper = shallowWrappperPdm(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call setMessage on pdm click', () => {
    const wrapper = shallowWrappperPdm(props);

    const spy = jest.spyOn(wrapper.instance(), "onPdmSelection");
    wrapper.instance().forceUpdate();

    wrapper.find('Link.at_0').simulate('click');
    expect(spy).toHaveBeenCalled();
  });
});
