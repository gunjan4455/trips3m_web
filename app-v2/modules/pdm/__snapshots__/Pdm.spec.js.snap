// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pdm: Render Correctly  should match with the previous snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pdm
    pdmsData={
      Object {
        "Hotel": Array [
          Object {
            "category": "Hotel",
            "message_enum": "PDM_CHANGE_HOTEL_NEED_BETTER_BUDGET",
            "message_text": "Hey! I need hotel in a better budget.",
            "pdm_text": "Need hotel in a better budget",
          },
          Object {
            "category": "Hotel",
            "message_enum": "PDM_CHANGE_HOTEL_NEED_BETTER_REVIEWS",
            "message_text": "Hey! I need a hotel with better reviews.",
            "pdm_text": "Need a hotel with better reviews",
          },
          Object {
            "category": "Hotel",
            "message_enum": "PDM_CHANGE_HOTEL_NEED_HIGHER_ROOM_CATEGORY",
            "message_text": "Hey! I need a hotel with a higher room category.",
            "pdm_text": "Need a hotel with a higher room category",
          },
          Object {
            "category": "Hotel",
            "message_enum": "PDM_BETTER_ROOM_SAME_HOTEL",
            "message_text": "Hey! I need a better room in the same hotel.",
            "pdm_text": "Need a better room in the same hotel",
          },
          Object {
            "category": "Hotel",
            "message_enum": "PDM_CHANGE_HOTEL_NEAR_SIGHTSEEING_AREA",
            "message_text": "Hey! I need a hotel near the market/sightseeing areas.",
            "pdm_text": "Need a hotel near the market/sightseeing areas",
          },
        ],
        "Inclusion": Array [
          Object {
            "category": "Inclusion",
            "message_enum": "PDM_ADD_INCLUSION_CONTACT_ME",
            "message_text": "Hey! I need to get a few inclusions added. Please contact me to discuss further.",
            "pdm_text": "Need to get an inclusion added",
          },
          Object {
            "category": "Inclusion",
            "message_enum": "PDM_REMOVE_INCLUSION_CONTACT_ME",
            "message_text": "Hey! I need to get a few inclusions removed. Please contact me to discuss further.",
            "pdm_text": "Need to get an inclusion removed",
          },
          Object {
            "category": "Inclusion",
            "message_enum": "PDM_MODIFY_INCLUSION_CONTACT_ME",
            "message_text": "Hey! I need modifications in a few inclusions. Please contact me to discuss further.",
            "pdm_text": "Need to modify an inclusion",
          },
        ],
        "Itinerary": Array [
          Object {
            "category": "Itinerary",
            "message_enum": "PDM_CHANGE_ITINERARY_DAYS_ORDER",
            "message_text": "Hey! I want to change the order of itinerary days.",
            "pdm_text": "I want to change the order of itinerary days",
          },
          Object {
            "category": "Itinerary",
            "message_enum": "PDM_CHANGE_ITINERARY_CITIES",
            "message_text": "Hey! I want to change the cities in the itinerary.",
            "pdm_text": "I want to change the cities in the itinerary",
          },
          Object {
            "category": "Itinerary",
            "message_enum": "PDM_MODIFY_ITINERARY_PARTICULAR_DAY",
            "message_text": "Hey! I want modification in itinerary of a particular day.",
            "pdm_text": "I want modification in itinerary of a particular day",
          },
          Object {
            "category": "Itinerary",
            "message_enum": "PDM_MODIFY_ITINERARY_ATTRACTIONS_COVERED",
            "message_text": "Hey! I want modification in the attractions being covered.",
            "pdm_text": "I want modification in the attractions being covered",
          },
        ],
      }
    }
    selectedPdm="Inclusion"
    setMessage={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ul>
        <li>
          <Link
            className="at_0"
            onClick={[Function]}
            onlyActiveOnIndex={false}
            style={Object {}}
          >
            Need to get an inclusion added
          </Link>
        </li>
        <li>
          <Link
            className="at_1"
            onClick={[Function]}
            onlyActiveOnIndex={false}
            style={Object {}}
          >
            Need to get an inclusion removed
          </Link>
        </li>
        <li>
          <Link
            className="at_2"
            onClick={[Function]}
            onlyActiveOnIndex={false}
            style={Object {}}
          >
            Need to modify an inclusion
          </Link>
        </li>
      </ul>,
      "className": "clearfix relative",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <Link
              className="at_0"
              onClick={[Function]}
              onlyActiveOnIndex={false}
              style={Object {}}
            >
              Need to get an inclusion added
            </Link>
          </li>,
          <li>
            <Link
              className="at_1"
              onClick={[Function]}
              onlyActiveOnIndex={false}
              style={Object {}}
            >
              Need to get an inclusion removed
            </Link>
          </li>,
          <li>
            <Link
              className="at_2"
              onClick={[Function]}
              onlyActiveOnIndex={false}
              style={Object {}}
            >
              Need to modify an inclusion
            </Link>
          </li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="at_0"
              onClick={[Function]}
              onlyActiveOnIndex={false}
              style={Object {}}
            >
              Need to get an inclusion added
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Need to get an inclusion added",
              "className": "at_0",
              "onClick": [Function],
              "onlyActiveOnIndex": false,
              "style": Object {},
            },
            "ref": null,
            "rendered": "Need to get an inclusion added",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="at_1"
              onClick={[Function]}
              onlyActiveOnIndex={false}
              style={Object {}}
            >
              Need to get an inclusion removed
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Need to get an inclusion removed",
              "className": "at_1",
              "onClick": [Function],
              "onlyActiveOnIndex": false,
              "style": Object {},
            },
            "ref": null,
            "rendered": "Need to get an inclusion removed",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="at_2"
              onClick={[Function]}
              onlyActiveOnIndex={false}
              style={Object {}}
            >
              Need to modify an inclusion
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Need to modify an inclusion",
              "className": "at_2",
              "onClick": [Function],
              "onlyActiveOnIndex": false,
              "style": Object {},
            },
            "ref": null,
            "rendered": "Need to modify an inclusion",
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "__emotion_base": "div",
      "__emotion_forwardProp": undefined,
      "__emotion_real": [Circular],
      "__emotion_styles": Array [
        Object {
          "& ul": Object {
            "& li": Object {
              "& a": Object {
                "&:hover": Object {
                  "color": "#009688",
                },
                "border": "solid 1px #009688",
                "borderRadius": "30px",
                "color": "#009688",
                "display": "block",
                "fontWeight": "400",
                "padding": "4px",
                "textAlign": "center",
                "textDecoration": "none",
              },
              "&:last-child": Object {
                "marginBottom": "0",
              },
              "display": "block",
              "listStyle": "none",
              "marginBottom": "8px",
            },
            "margin": "0",
            "padding": "0",
          },
          "background": "#ffffff",
          "boxShadow": "0 -2px 20px 0 rgba(0, 0, 0, 0.2)",
          "height": "auto",
          "padding": "8px 8px",
          "width": "100%",
        },
      ],
      "defaultProps": undefined,
      "displayName": "Styled(div)",
      "render": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ul>
          <li>
            <Link
              className="at_0"
              onClick={[Function]}
              onlyActiveOnIndex={false}
              style={Object {}}
            >
              Need to get an inclusion added
            </Link>
          </li>
          <li>
            <Link
              className="at_1"
              onClick={[Function]}
              onlyActiveOnIndex={false}
              style={Object {}}
            >
              Need to get an inclusion removed
            </Link>
          </li>
          <li>
            <Link
              className="at_2"
              onClick={[Function]}
              onlyActiveOnIndex={false}
              style={Object {}}
            >
              Need to modify an inclusion
            </Link>
          </li>
        </ul>,
        "className": "clearfix relative",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <Link
                className="at_0"
                onClick={[Function]}
                onlyActiveOnIndex={false}
                style={Object {}}
              >
                Need to get an inclusion added
              </Link>
            </li>,
            <li>
              <Link
                className="at_1"
                onClick={[Function]}
                onlyActiveOnIndex={false}
                style={Object {}}
              >
                Need to get an inclusion removed
              </Link>
            </li>,
            <li>
              <Link
                className="at_2"
                onClick={[Function]}
                onlyActiveOnIndex={false}
                style={Object {}}
              >
                Need to modify an inclusion
              </Link>
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="at_0"
                onClick={[Function]}
                onlyActiveOnIndex={false}
                style={Object {}}
              >
                Need to get an inclusion added
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Need to get an inclusion added",
                "className": "at_0",
                "onClick": [Function],
                "onlyActiveOnIndex": false,
                "style": Object {},
              },
              "ref": null,
              "rendered": "Need to get an inclusion added",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="at_1"
                onClick={[Function]}
                onlyActiveOnIndex={false}
                style={Object {}}
              >
                Need to get an inclusion removed
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Need to get an inclusion removed",
                "className": "at_1",
                "onClick": [Function],
                "onlyActiveOnIndex": false,
                "style": Object {},
              },
              "ref": null,
              "rendered": "Need to get an inclusion removed",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="at_2"
                onClick={[Function]}
                onlyActiveOnIndex={false}
                style={Object {}}
              >
                Need to modify an inclusion
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Need to modify an inclusion",
                "className": "at_2",
                "onClick": [Function],
                "onlyActiveOnIndex": false,
                "style": Object {},
              },
              "ref": null,
              "rendered": "Need to modify an inclusion",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "__emotion_base": "div",
        "__emotion_forwardProp": undefined,
        "__emotion_real": [Circular],
        "__emotion_styles": Array [
          Object {
            "& ul": Object {
              "& li": Object {
                "& a": Object {
                  "&:hover": Object {
                    "color": "#009688",
                  },
                  "border": "solid 1px #009688",
                  "borderRadius": "30px",
                  "color": "#009688",
                  "display": "block",
                  "fontWeight": "400",
                  "padding": "4px",
                  "textAlign": "center",
                  "textDecoration": "none",
                },
                "&:last-child": Object {
                  "marginBottom": "0",
                },
                "display": "block",
                "listStyle": "none",
                "marginBottom": "8px",
              },
              "margin": "0",
              "padding": "0",
            },
            "background": "#ffffff",
            "boxShadow": "0 -2px 20px 0 rgba(0, 0, 0, 0.2)",
            "height": "auto",
            "padding": "8px 8px",
            "width": "100%",
          },
        ],
        "defaultProps": undefined,
        "displayName": "Styled(div)",
        "render": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
